### PR TITLE
Add Table Expiry PR Comments

### PR DESCRIPTION
**Pull request by vedantchokshi**
Had to make some big changes from the PR comments so putting up another PR so the differences can be seen.

Most of the files are renames but the biggest change is the way the MessageEventHandler works now. It has been changes to use a generator.

Fixes #

### CODE REVIEW

File: beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/service/PagingCleanupServiceTest.java [hunk 1]
```diff
@@ -95,8 +98,8 @@ public void typicalWithPaging() {
   public void mixOfScheduledAndFailedPaths() {
     UnreferencedHandler handler = new UnreferencedHandler(housekeepingPathRepository, pathCleaner);
     pagingCleanupService = new PagingCleanupService(List.of(handler), 2, false);
-    List<EntityHousekeepingPath> paths = List.of(
-        createEntityHousekeepingPath("s3://some_foo", HousekeepingStatus.SCHEDULED),
+    List<HousekeepingPath> paths = List.of(
+        createEntityHousekeepingPath("s3://some_foo", SCHEDULED),
         createEntityHousekeepingPath("s3://some_bar", HousekeepingStatus.FAILED)
```
Comment (JayGreeeen) [inline beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/service/PagingCleanupServiceTest.java:103]: Static import `FAILED` too. Also on line 120 and 170-172.

File: README.md [hunk 2]
```diff
@@ -199,7 +198,7 @@ The following table shows the configuration that can be provided:
 
 ## External links
 
-Please see the [Housekeeping](https://github.com/HotelsDotCom/housekeeping) library for more information.
+Please see the [Housekeeping](https://github.com/HotelsDotCom/housekeepingEntity) library for more information.
```
Comment (abhimanyugupta07) [inline README.md:201]:
```suggestion
Please see the [Housekeeping](https://github.com/HotelsDotCom/housekeeping) library for more information.
```

File: beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/model/HousekeepingPath.java [hunk 3]
```diff
@@ -50,9 +50,9 @@
   @Column(name = "table_name")
   private String tableName;
 
-  @Column(name = "path_status", nullable = false)
```
Comment (patduin) [inline beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/model/HousekeepingPath.java:53]: So this is gonna require a migration script + documentation how to move from version 2 to 3.

File: beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/generator/UnreferencedHousekeepingPathGenerator.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  private static final String UNREFERENCED_DATA_RETENTION_PERIOD_PROPERTY_KEY = "beekeeper.unreferenced.data.retention.period";
+  private final CleanupDelayExtractor cleanupDelayExtractor;
+  private final Clock clock;
+
+  public UnreferencedHousekeepingPathGenerator(String cleanupDelay) {
+    this(new CleanupDelayExtractor(UNREFERENCED_DATA_RETENTION_PERIOD_PROPERTY_KEY, cleanupDelay),
+        Clock.systemDefaultZone());
+  }
+
+  UnreferencedHousekeepingPathGenerator(CleanupDelayExtractor cleanupDelayExtractor, Clock clock) {
```
Comment (patduin) [inline beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/generator/UnreferencedHousekeepingPathGenerator.java:57]: @VisibleForTesting just to be consistent

File: beekeeper-scheduler-apiary/src/main/resources/db/migration/V2_1__Create_unpartitioned_table_housekeeping_table_and_alter_path_table_name_and_rename_path_status_column.sql [hunk 5]
```diff
@@ -13,3 +14,6 @@ CREATE TABLE IF NOT EXISTS table_housekeeping (
   lifecycle_type varchar(255) NOT NULL,
   PRIMARY KEY (id)
 );
+
+ALTER TABLE path RENAME TO housekeeping_path;
```
Comment (patduin) [inline beekeeper-scheduler-apiary/src/main/resources/db/migration/V2_1__Create_unpartitioned_table_housekeeping_table_and_alter_path_table_name_and_rename_path_status_column.sql:18]: ah found it! Nevermind my earlier comment.

File: beekeeper-scheduler-apiary/src/test/java/com/expediagroup/beekeeper/scheduler/apiary/generator/utils/CleanupDelayExtractorTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  @Test
+  public void typicalExtractCleanUpDelay() {
+    String cleanupDelayInput = "P30D";
+    when(listenerEvent.getTableParameters()).thenReturn(Map.of(EXTRACTOR_KEY, cleanupDelayInput));
+    Duration cleanupDelay = cleanupDelayExtractor.extractCleanupDelay(listenerEvent);
+    assertThat(cleanupDelay).isEqualTo(Duration.parse(cleanupDelayInput));
+  }
+
+  @Test
+  public void NoKeyExtractCleanUpDelay() {
```
Comment (patduin) [inline beekeeper-scheduler-apiary/src/test/java/com/expediagroup/beekeeper/scheduler/apiary/generator/utils/CleanupDelayExtractorTest.java:55]: N=noKey lowercase

**Pull request merged**